### PR TITLE
Fix various typos

### PR DESCRIPTION
**Pull request by luzpaz**
The PR contains some source code changes that break API. The intention of the PR is to highlight these typos and hopefully they can get addressed and made backward compatible if necessary.

### CODE REVIEW

**Review by TurboGit — APPROVED** [review]
Thanks!

File: src/libs/location.c [hunk 1]
```diff
@@ -616,15 +616,15 @@ static void _lib_location_parser_start_element(GMarkupParseContext *cxt, const c
...
         else if(strcmp(*avalue, "hamlet") == 0)
           place->type = LOCATION_TYPE_HAMLET;
         else if(strcmp(*avalue, "city") == 0)
           place->type = LOCATION_TYPE_CITY;
         else if(strcmp(*avalue, "administrative") == 0)
           place->type = LOCATION_TYPE_ADMINISTRATIVE;
-        else if(strcmp(*avalue, "residental") == 0)
-          place->type = LOCATION_TYPE_RESIDENTAL;
+        else if(strcmp(*avalue, "residential") == 0)
+          place->type = LOCATION_TYPE_RESIDENTIAL;
```
Comment (luzpaz) [inline src/libs/location.c:627]: I believe there needs to be code here that can digest both `residental` (backward compatible) and `residential`

Comment (TurboGit) [reply src/libs/location.c:627]: Good point indeed! I missed that, Can you propose a PR?

Comment (TurboGit) [reply src/libs/location.c:627]: @luzpaz : I think this even need to be reverted. The term "residental" exists and seems to come from an external library. So better revert all RESIDENTIAL back to RESIDENTAL and wait for @phweyland comment. TIA.

Comment (phweyland) [reply src/libs/location.c:627]: `residental` seems to be discouraged by openstreetmap but the tag exists anyway (16692 times):
image
Map creators should use `residential`;
So the above @luzpaz suggestion sounds good.
Thanks

Comment (luzpaz) [reply src/libs/location.c:627]: please advice on how to proceed? (if needed?)

Comment (TurboGit) [reply src/libs/location.c:627]: Given @phweyland answer I would add a check for both `"residential"` and `"residental"` as alias and set `place->type = LOCATION_TYPE_RESIDENTIAL;`.

Comment (luzpaz) [reply src/libs/location.c:627]: See #11168

**Pull request merged**